subsys/net/lib/lwm2m/lwm2m_rd_client.c

```diff
@@ -373,13 +394,17 @@ int engine_trigger_bootstrap(void)
 #if defined(CONFIG_LWM2M_RD_CLIENT_SUPPORT_BOOTSTRAP)
 	if (!sm_is_registered()) {
 		/* Bootstrap is not possible to trig */
+		k_mutex_lock(&state_lock, K_FOREVER);
 		LOG_WRN("Cannot trigger bootstrap from state %u", client.engine_state);
+		k_mutex_unlock(&state_lock);
 		return -EPERM;
 	}
 
 	LOG_INF("Server Initiated Bootstrap");
 	client.use_bootstrap = true;
+	k_mutex_lock(&state_lock, K_FOREVER);
 	client.engine_state = ENGINE_INIT;
+	k_mutex_unlock(&state_lock);
 
```

Comment:
I believe Zephyr mutexes are reentrant so we could lock multiple times, therefore this function and similar could be written like:
```suggestion
        k_mutex_lock(&state_lock, K_FOREVER);
	if (!sm_is_registered()) {
		/* Bootstrap is not possible to trig */
		LOG_WRN("Cannot trigger bootstrap from state %u", client.engine_state);
		k_mutex_unlock(&state_lock);
		return -EPERM;
	}

	LOG_INF("Server Initiated Bootstrap");
	client.use_bootstrap = true;
	client.engine_state = ENGINE_INIT;
	k_mutex_unlock(&state_lock);

```